saleor/order/actions.py

```diff
@@ -910,19 +908,36 @@ def create_refund_fulfillment(
     unfulfilled lines will be deallocated.
     """
 
-    shipping_refund_amount = __get_shipping_refund_amount(
-        refund_shipping_costs, amount, order.shipping_price_gross_amount
-    )
-
     with transaction_with_commit_on_errors():
+        refund_shipping_costs = False
+        # Amount to be eventually specified in the Fulfillment object.
+        counted_shipping_amount = None
+        for item in payments:
+            if item["include_shipping_costs"]:
+                if not item["amount"]:
+                    counted_shipping_amount = order.shipping_price_gross_amount
+                refund_shipping_costs = True
+                shipping_refund_amount = __get_shipping_refund_amount(
```

Comment:
Will it work as expected? `__get_shipping_refund_amount` sets amount to refund only when `item["amount"]` will be None and you build a payment_to_refund structure you set `"amount": amount_to_refund or Decimal("0"),` 
fulfillments.py line 775
https://github.com/mirumee/saleor/pull/8165/files#diff-bb17901778348f1cf6d54c9c0b102b8e7ea967d9df62385315c45c786d93ffc9R735